guava/src/main/kotlin/com/spectralogic/kotlinutils/core/curry.kt

```diff
@@ -0,0 +1,18 @@
+package com.spectralogic.kotlinutils.core
```

Comment:
The filename for this should be Capitalized.

Reply:
Also, should this file be in the guava module, or in a core module?

Reply:
My thinking was that curry doesn't rely on Guava, so it should be in core, but I can move it.